ci/generate-workflows.py

```diff
@@ -325,25 +325,38 @@ def install_git(self):
 
         return steps
 
-    def install_rust(self, cache=True):
+    def install_rust(self, cache=True, toolchain="stable"):
         salt = "2"
         key_prefix = f"{self.name}-{self.rust_target}-{salt}-${{{{ runner.os }}}}-${{{{ hashFiles('**/Cargo.lock') }}}}"
-        params = {
-            "profile": "minimal",
-            "toolchain": "stable",
-            "override": True,
-            "components": "rustfmt",
-        }
+        params = dict()
         if self.rust_target:
             params["target"] = self.rust_target
         steps = [
             ActionStep(
                 name="Install Rust",
-                action="actions-rs/toolchain@v1",
+                action=f"dtolnay/rust-toolchain@{toolchain}",
                 params=params,
-                env={"ACTIONS_ALLOW_UNSECURE_COMMANDS": "true"},
             ),
         ]
+        if "centos7" in self.name:
```

Comment:
I think this conditional should have an else and that the ActionStep from above should go in the else?

Reply:
Addressed in latest commit, I don't think it looks much better...

Reply:
it's not about how it looks so much as not trying to run the action that will fail on centos7.

Did you re-run this and amend in the workflows? GH is only showing me this file as being change in the latest update to the PR?

Reply:
There is no change to the actions when rerun, it's a very tiny thing, thanks Python, but originally the `if` replaces steps `=`, instead of appending to them `+=`, the change is semantically the same...

Reply:
Ah, subtle. I didn't notice it was replacing, which is why I asked for the `if` to be moved